include/arbor/gpu_context.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <memory>
+
+#include <arbor/version.hpp>
+
+namespace arb {
+struct gpu_context {
+    bool has_gpu_;
+    size_t attributes_;
+    gpu_context();
+
+#ifdef ARB_GPU_ENABLED
+    bool has_concurrent_managed_access();
+    bool has_atomic_double();
+#endif
+};
+}
```

Comment:
Rather than have the GPU-dependent code in the public header, we should arrange things such that to users of the code, the context is opaque (e.g. just a forward declaration `struct gpu_context`) — it might not even need its own header, instead being forward-declared in `execution_context.hpp`.

Then the actual contents of the `gpu_context` structure would be present in a private library header for consumption by the library code. The `#ifdef` guard would then check `ARB_HAVE_GPU` as before.